common/predictive-text/virtualizedWorker.ts

```diff
@@ -0,0 +1,86 @@
+/// <reference path="embedded_worker.d.ts" />
+/// <reference path="worker-interface.d.ts" />
+
+namespace com.keyman.text.prediction {
+  // Unfortunately, importing like this loses all type information for the modules.
+  // It's that, or massively overhaul lm-layer to be module-based instead of
+  // namespace-based.
+  var fs = require("fs");
+  var vm = require("vm");
+
+  class VirtualizedWorkerContext {
+    // The LMLayerWorker installs itself to 'self', the expected Worker global, so we provide an alias.
+    self: VirtualizedWorkerContext;
+
+    constructor() {
+      this.self = this;
+    }
+
+    postMessage: (message: any, options: PostMessageOptions) => void;
+
+    importScripts() {
+      /* Use of vm.createContext and script.runInContext allow us to avoid
+        * polluting the global scope with imports.  When we throw away the
+        * context object, imported scripts will be automatically GC'd.
+        */
+      for(var i=0; i < arguments.length; i++) {
+        this.__importScriptString(fs.readFileSync(arguments[i]) as any as string);
+      }
+    }
```

Comment:
Doesn't `fs.readFileSync()` without an `encoding` parameter [return `Buffer` by default[(https://nodejs.org/api/fs.html#fs_fs_readfilesync_path_options)? Either way, I'm suggesting these changes to better take advantage of TypeScript features, and have less uses of `as any as string` typing hacks.

```suggestion
    importScripts(...scriptNames: string[]) {
      /* Use of vm.createContext and script.runInContext allow us to avoid
       * polluting the global scope with imports.  When we throw away the
       * context object, imported scripts will be automatically GC'd.
       */
      for (let script of scriptNames) {
        this.__importScriptString(fs.readFileSync(script, "UTF-8"));
      }
    }
```